README.md

```diff
@@ -210,3 +211,261 @@ developer console](https://code.google.com/apis/console/).
+* 400 if some properties are missing or invalid.
+* 403 if `oldPassword` is incorrect.
+* 404 if no reminder with this ID exists or if the logged-in user isn't this
+  user.
+
+### DELETE `/users/{id}`
+
+Required: the header `Authorization` that identifies a user. A user can only be
+deleted by himself.
+
+#### Output
+* 204 if request succeeded.
+* 404 if no reminder with this ID exists or if the logged-in user isn't this
+  user.
+
+## API groups
+
+This API manages groups of users.
+
+### POST `/groups`
+
+Required: the header `Authorization` that identifies a user.
+
+This creates a new group.
+
+#### Input
+
+All properties must be present. There is no default value. For example:
+```json
+{
+  "name": "Holmes"
+}
+```
+
+#### Output
+* 201 if request succeeded, with `location` header indicating the new resource URL.
+* 400 if some properties are missing or invalid.
+
+### GET `/groups/{id}`
+
+Required: the header `Authorization` that identifies a user. Only a user that
+belongs to this group can access its information.
+
+This returns information about a group.
+
+#### Output
+* 200 if request succeeded, with body:
+```json
+{
+  "name": "Holmes"
+}
+```
+* 404 if no group with this ID exists or if the user doesn't belong to this group.
+
+### GET `/groups/{id}/members`
+Required: the header `Authorization` that identifies a user. Only a user that
+belongs to this group can see its users.
+
+This returns information about users belonging to this group.
+
+#### Output
+* 200 if request succeeded, with body:
+```json
+[
+  {
+    "id": 1,
+    "forename": "Sherlock",
+    "email": "Sherlock.Holmes@baker-street.co.uk",
+    "isAdmin": true,
+    "location": "https://server.name/api/v1/users/1"
+  },
+  {
+    ...
+  }
+]
+```
+  `isAdmin` is true if this user is an admin for this group.
+* 404 if no user with this ID exists or if the user doesn't belong to this group.
+
+### PUT `/groups/{id}/members/{userId}`
+Required: the header `Authorization` that identifies a user. A group can only be
+modified by one of its admin user.
+
+This will add the user with id `userId` to this group.
+
+### Output
+
+* 204 if request succeeded.
+* 403 if this logged-in user is not an admin for this group.
+* 404 if no group with this ID exists or if the logged-in user is not part of
+  this group.
+
+### DELETE `/groups/{id}/members/{userId}`
+
+Required: the header `Authorization` that identifies a user. A group can only be
+modified by one of its admin user.
+
+This will remove the user with id `userId` from this group.
+
+### Output
+
+* 204 if request succeeded.
+* 403 if this logged-in user is not an admin for this group.
+* 404 if no group with this ID exists or if the logged-in user is not part of
+  this group.
+
+### PUT `/groups/{id}`
+
+Required: the header `Authorization` that identifies a user. A group can only be
+modified by one of its admin user.
+
+#### Input
+
+All properties must be present. For example:
+
+```json
+{
+  "name": "Moriarty"
```

Comment:
"Moriarty" is a family name ;) "James" is the first name. but if it's not obvious enough I can change it.

Maybe "Good_Guys" ;)